### DISCUSSION TITLE
Conditional rendering in a page when building a form & Populating the form based on the data from a query lookup

### DISCUSSION THREAD

**Discussion by rwxzig** [OrchardCMS/OrchardCore#15126] [Q&A] [answered]
1. Is there any way to render different inputs based on some conditions, for instance if we have an input of type checkbox, depending if that checkbox is ticked or not, we render different part of the form?
2. Suppose that I have a form where I require the user to enter: name, birthday, address, phone nr etc,  is there any way to do a lookup and populate the form with the data that we already have on the user and only require the user to enter the data that is missing?

I am looking at OrchardCore.Forms for the second, and I think we would need to extend the usability of the module, but could you give some guidance on how would be best to achieve this?

**Comment by Piedone** [comment]
Are you looking to do this specifically with the Forms module? Because for anything more than really simple forms (and especially if you can write code), I'd recommend creating these forms in code. Then, you can render forms in any way you'd prefer.

**Reply by rwxzig** [reply]
Do you mean that to write raw html from types in that forms support? Or is somewhere else where to write the forms so that they can be rendered to a page?

**Reply by Piedone** [reply, marked as answer]
I meant the standard MVC way. But if you want to include a form in a content item then embedding it as Liquid (with e.g. a Liquid widget) or by displaying a custom widget that displays it from Razor would work too.
